core/lib/mini_merkle_tree/src/lib.rs

```diff
@@ -86,67 +120,146 @@ where
             hasher,
             hashes,
             binary_tree_size,
+            start_index: 0,
+            cache: vec![None; depth],
         }
     }
 
+    /// Returns `true` if the tree is empty.
+    pub fn is_empty(&self) -> bool {
+        self.start_index == 0 && self.hashes.is_empty()
+    }
+
     /// Returns the root hash of this tree.
     /// # Panics
-    /// Will panic if the constant below is invalid.
-    pub fn merkle_root(self) -> H256 {
+    /// Should never panic, unless there is a bug.
```

Comment:
Nit: No need to document a panic then; instead, put `#[allow(clippy::missing_panics_doc)]` on the method with a this comment.